src/components/modals/ButtonsModal.js

```diff
@@ -69,7 +71,12 @@ export function ButtonsModal<Buttons: { [key: string]: ButtonInfo }>(props: {|
       </View>
       <View style={styles.buttons}>
         {Object.keys(buttons).map((key, i) => {
-          const defaultType = i === 0 && Object.keys(buttons).length > 1 ? 'primary' : 'secondary'
+          let defaultType
+          if (theme.preferPrimaryButton) {
+            defaultType = i === 0 ? 'primary' : 'secondary'
+          } else {
+            defaultType = i === 0 && Object.keys(buttons).length > 1 ? 'primary' : 'secondary'
```

Comment:
Replace `Object.keys(buttons)` with third argument from the map function.

Reply:
done